File: CineVibe/CineVibe.Services/Database/Screening.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CineVibe.Services.Database
{
    public class Screening
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public DateTime StartTime { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        // Foreign keys
        [Required]
        public int MovieId { get; set; }
        
        [Required]
        public int HallId { get; set; }
        
        [Required]
        public int ScreeningTypeId { get; set; }
        
        // Navigation properties
        public virtual Movie Movie { get; set; } = null!;
        public virtual Hall Hall { get; set; } = null!;
        public virtual ScreeningType ScreeningType { get; set; } = null!;
    }
}

File: CineVibe/UI/cinevibe_mobile/lib/model/screening.dart
import 'package:json_annotation/json_annotation.dart';

part 'screening.g.dart';

@JsonSerializable()
class Screening {
  final int id;
  final DateTime startTime;
  final bool isActive;
  final DateTime createdAt;
  final int movieId;
  final String movieTitle;
  final int movieDuration;
  final int hallId;
  final String hallName;
  final int screeningTypeId;
  final String screeningTypeName;
  final double price;
  final DateTime endTime;
  final int occupiedSeatsCount;

  const Screening({
    required this.id,
    required this.startTime,
    required this.isActive,
    required this.createdAt,
    required this.movieId,
    required this.movieTitle,
    required this.movieDuration,
    required this.hallId,
    required this.hallName,
    required this.screeningTypeId,
    required this.screeningTypeName,
    required this.price,
    required this.endTime,
    required this.occupiedSeatsCount,
  });

  factory Screening.fromJson(Map<String, dynamic> json) => _$ScreeningFromJson(json);
  Map<String, dynamic> toJson() => _$ScreeningToJson(this);
}
